recipe.cake

```diff
@@ -13,7 +13,9 @@ BuildParameters.SetParameters(context: Context,
                             shouldRunDupFinder: false,
                             shouldRunInspectCode: false,
                             shouldRunDotNetCorePack: true,
-                            shouldUseDeterministicBuilds: true);
+                            shouldUseDeterministicBuilds: true,
+                            preferredBuildAgentOperatingSystem: PlatformFamily.Windows,
+                            preferredBuildProviderType: BuildProviderType.GitHubActions);
```

Comment:
You may not really want to switch to GitHubActions when doing publishing.
From my own experience, I have had problems with using GitHubActions + GitVersion on tagged releases.

Most likely related to the following issue: https://github.com/GitTools/GitVersion/issues/2074

Reply:
@gep13 any insights here? I've only used MinVer with GH actions. There I needed to set the depth when cloning the repo to include everything https://github.com/mholo65/depends/blob/master/.github/workflows/CI.yml#L17

Reply:
Hmm, I will have a proper read through that thread later to see if there is anything that can be done.

Reply:
GitVersion have the same requirement. But I believe using `git fetch --prune --unshallow` is preferred instead.

However, the problem is the issue I linked to is not related to that.

Basically, if you run a build on the tag `1.0.0` you would end up with a version similar to `1.1.0-tags-1.0.0-alpha.1.1` (may not be exactly that version, but would be something similar).
It seems to be due to a bug related to the normalization (as mentioned in the linked issue).

Reply:
@mholo65 if you wanted to remove GitVersion from the equation, you have the option of not using it.  This is documented here:

https://cake-contrib.github.io/Cake.Recipe/docs/extending/override-generating-version

Reply:
just an FYI, if you override the version generation, and the tool you use does not update the appveyor version or do not use the same format (assuming you will continue to use it), you must also replace the codecov task as well (or disable).

As an example see: https://github.com/cake-contrib/Cake.Warp/blob/a87ef77d7bb717cecc1bc491d8ad81ee3a73c201/.build/codecov.cake that I am working on replacing GitVersion with my own versioning tool (that do not touch AppVeyor version at all).

Reply:
@admiriingworm is that some information that you could capture in the Cake.Recipe docs?